go/felix/intdataplane/status_combiner.go

```diff
@@ -84,16 +83,33 @@ func (e *endpointStatusCombiner) Apply() {
 		}
 		if statusToReport == "" {
 			logCxt.Info("Reporting endpoint removed.")
-			e.fromDataplane <- &proto.WorkloadEndpointStatusRemove{
-				Id: &id,
+			switch id := id.(type) {
```

Comment:
It's syntactic sugar, the new `id` variable shadows the outer one and it takes on the type of the particular `case` that it's in.